binder/views.py

```diff
@@ -1626,6 +1666,116 @@ def _store_field(self, obj, field, value, request, pk=None):
 								}
 							})
 					setattr(obj, f.attname, value)
+
+				elif isinstance(f, models.FileField):
+					# If the value is a str that matches how we return file
+					# fields we convert it to the correct file
+					if isinstance(value, str):
+						match = re.match(r'/api/(\w+)/(\d+)/(\w+)/', value)
+						model, pk_, field_ = match.groups()
+						try:
+							model = self.router.name_models[model]
+						except KeyError:
+							pass
+						else:
+							view = self.get_model_view(model)
+							if field_ in view.file_fields:
+								try:
+									obj_ = view.get_queryset(request).get(pk=pk_)
+								except model.DoesNotExist:
+									pass
+								else:
+									value = getattr(obj_, field_)
+
+					if not isinstance(value, File) and value is not None:
+						raise BinderFieldTypeError(self.model.__name__, field)
+
+					if value is not None:
+						if value.size > self.max_upload_size * 10**6:
+							raise BinderFileSizeExceeded(self.max_upload_size)
+
+						if getattr(f, 'allowed_extensions', None) is not None:
+							extension = None if '.' not in value.name else value.name.split('.')[-1]
+
+							if extension not in f.allowed_extensions:
+								raise BinderFileTypeIncorrect([{'extension': t} for t in f.allowed_extensions])
+
+						if isinstance(f, (models.ImageField, BinderImageField)):
+							try:
+								img = Image.open(value)
```

Comment:
Could you add  that if we have an ImageField, and no allowed extensions are specified,  that we check on the extension before parsing the image? This is a security issue otherwise